### PR TITLE
Many build improvements, fixing `clang` and OpenSSL version and CMake `find_package()` issues

### PR DESCRIPTION
**Pull request by rajeev-0**
Among others, fix CMake `find_package()` occasionally incorrectly reporting error in finding OpenSSL.

* README.md: require at least OpenSSL version 3.0
* files.c: remove incompatible #define OSSL_DEPRECATEDIN_3_1
* Makefile_v1: add -Werror only if NDEBUG is defined
* CMakeLists.txt: add workaround for using non-installed OpenSSL builds with default libdir
* Makefile_v1,README.md: major update with improved diagnostics
* src/CMakeLists.txt: remove -Werror. It is enabled only for development and CI, using Makefile_v1 without NDEBUG
* CMakeLists.txt,REAME.md: improve use and diagnostics of OpenSSL package detection
* Makefile_v1: clean target: avoid errors trying to remove files not (anymore) existing
* src/CMakeLists.txt,README.md: add support for clang version 19.1.1 (adding -Wno-gnu-folding-constant)

### CODE REVIEW

**Comment by DDvO** [comment]
This meanwhile provides more thorough fixes for various build issues also with `Makefile_v1`, 
extending also the diagnostics on finding the OpenSSL include directory and libraries.

**Comment by DDvO** [comment]
@benjamin-schilling and @rajeev-0 finally, really ready for review.

**Review by DDvO — APPROVED** [review]

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -3,11 +3,12 @@ target_compile_options(secutils_compilation_options INTERFACE
     -Wall -Woverflow -Wextra -Wswitch -Wmissing-prototypes -Wstrict-prototypes
     -Wformat -Wformat-security -Wtype-limits -Wundef
     -Wsign-compare -Wpointer-arith -Wunused-parameter
-    -pedantic -Werror
+    -pedantic
+    # -Werror is enabled only for development and CI using Makefile_v1 without NDEBUG
```
Comment (benjamin-schilling) [inline src/CMakeLists.txt:7]: Why that?

Comment (DDvO) [reply src/CMakeLists.txt:7]: I want to do this for the genCMPClient and libcmp and propose to do this also for libSecUilts
due to experience of users trying to build the libraries using newer/other compilers (e.g., from other OS variants such as Mac).
These tools can be more picky/pedantic on minor potential issues than the compilers typically used.
The idea is to let such users get past small issues that may lead to unexpected warnings, without getting the build aborted.
They will still see the warnings and then can decide for themselves to ignore them or not.

Anyway, the CI runs continue using `-Werror` with `Makefile_v1` such that all warnings emitted
by the typical compilers lead to CI failure and in this way will bring all compiler-reported issues to the maintainers' attention.

Yet if you feel uncomfortable with this change, I can 
* undo it for libSecUtils in general or
* undo it for the case that CMake is used with debug builds (such that `-Werror` would be kept for debug builds both using `Makefile_v1` and CMake) - 

as you prefer.

**Comment by DDvO** [comment]
Thank you @benjamin-schilling

**Pull request merged**